web/src/components/datatable.tsx

```diff
@@ -57,7 +58,9 @@ export function DataTable<T>(props: DataTableProps<T>): JSX.Element {
         <div class='flex-grow' />
         {props.extraButtons}
         <Show when={props.refresh}>
-          <Button onClick={() => props.refresh?.()}>Refresh</Button>
+          <Button onClick={() => props.refresh?.()}>
+            {translate('refresh')}
+          </Button>
```

Comment:
Might be worth to handle translation inside the button it self. This way we also force us to add translations for new button actions / texts.

Reply:
That creates a whole bunch of ambiguities, 1) not everything will be in a special purpose container, 2) we'd need to keep track of which containers force i18n and which don't, 3) i18nizing containers become "terminal" and can't be composed further.

I also kind of like that this makes it easy to grep for `translate(` if you want to find all translation strings (probably something we'll want to look tooling for at some point to track).

Reply:
Your points are solid. Let's use the current approach.